### PR TITLE
Display for Rope.

### PR DESCRIPTION
**Pull request by sjoshid**
It looks like there has been no real update for #772. @cmyr is this what you were looking for?
I can write few test cases too. But first wanted to know if this is acceptable.

### CODE REVIEW

File: rust/rope/src/rope.rs [hunk 1]
```diff
@@ -730,6 +730,16 @@ impl<'a> From<&'a Rope> for String {
     }
 }
 
+impl fmt::Display for Rope {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut ci = self.iter_chunks_all();
+        for s in ci.next() {
+            write!(f, "{}", s).is_ok();
```
Comment (mqzry) [inline rust/rope/src/rope.rs]: You should branch on `f.alternate()` like the impl of `fmt::Debug` and the other pull request.

File: rust/rope/src/rope.rs [hunk 2]
```diff
@@ -730,6 +730,16 @@ impl<'a> From<&'a Rope> for String {
     }
 }
 
+impl fmt::Display for Rope {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut ci = self.iter_chunks_all();
```
Comment (mqzry) [inline rust/rope/src/rope.rs]: The function `iter_chunks_all` does not exist. Use `iter_chunks(0..self.len())` instead to get a iterator over the whole rope.

Comment (cmyr) [reply rust/rope/src/rope.rs]: Yea, iter_chunks_all was removed a few days ago, the replacement is `iter_chunks(..)`. (we can now specify ranges in these functions, which is convenient).

File: rust/rope/src/rope.rs (hunk 1 above)
Comment (cmyr) [reply rust/rope/src/rope.rs]: In my opinion an alternate representation isn't strictly necessary, that's more a fancy bonus feature.

One issue here though is that we want to propagate any error that occur in the `write!` macro; the `is_ok()` call basically just ignores them, we should be able to just do `write!(...)?;` and the error will be handled correctly.

File: rust/rope/src/rope.rs (hunk 2 above)
Comment (sjoshid) [reply rust/rope/src/rope.rs]: I thought I had my fork synced with remote but it wasnt. I have made changes in my latest commit.

File: rust/rope/src/rope.rs (hunk 1 above)
Comment (sjoshid) [reply rust/rope/src/rope.rs]: I read in @cmyr comment somewhere it isn't super important. Hence I ignored it.

File: rust/rope/src/rope.rs [hunk 3]
```diff
@@ -756,6 +756,16 @@ impl<'a> From<&'a Rope> for String {
     }
 }
 
+impl fmt::Display for Rope {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut ci = self.iter_chunks(..);
+        for s in ci.next() {
```
Comment (cmyr) [inline rust/rope/src/rope.rs]: my only other observation here is that in rust you would normally just write this as,

```rust
for s in self.iter_chunks(..) {
// do something
}
```

That is to say, you don't need the extra step of assigning the iterator to a variable, you can just use it directly.

Comment (sjoshid) [reply rust/rope/src/rope.rs]: Great. Will make that change.

**Comment by sjoshid** [comment]
I agree.

**Comment by cmyr** [comment]
Thanks for this!

**Pull request merged**